config/utils_methods.py

```diff
@@ -433,3 +433,200 @@ def validate_uuid(uuid_to_test, version=4):
 
 
         
+
+def get_related_data(model, serializer_class, filter_field, filter_value):
+        """
+        Retrieves related data for a given model, serializer, and filter field.
+        """
+        try:
+            related_data = model.objects.filter(**{filter_field: filter_value})
+            serializer = serializer_class(related_data, many=True)
+            logger.debug("Retrieved related data for model %s with filter %s=%s.",
+                        model.__name__, filter_field, filter_value)
+            return serializer.data
+        except Exception as e:
+            logger.exception("Error retrieving related data for model %s with filter %s=%s: %s",
+                            model.__name__, filter_field, filter_value, str(e))
+            return []
+
+     
+def format_phone_number(phone_number):
+    phone_number_str = str(phone_number) 
+    
+    # Check if the phone number length is correct
+    if len(phone_number_str) == 10:
+        return "91" + phone_number_str  #
+    elif len(phone_number_str) == 12 and phone_number_str.startswith("91"):
+        return phone_number_str  
+    else:
+        return "Mobile number has incorrect length"  
+
+from django.core.mail import EmailMessage
+def send_pdf_via_email(to_email, pdf_relative_path):
+    """Send the generated PDF as an email attachment."""
+    
+    # Construct the full path to the PDF file
+    pdf_full_path = os.path.join(MEDIA_ROOT, pdf_relative_path)
+    
+    subject = 'Your Sales Order Receipt'
+    body = 'Please find attached your sales order receipt.'
+    email = EmailMessage(subject, body, to=[to_email])
+
+    # Ensure the PDF file exists before attempting to open it
+    if not os.path.exists(pdf_full_path):
+        raise FileNotFoundError(f"The file {pdf_full_path} does not exist.")
+
+    # Read the PDF file from the provided full path
+    with open(pdf_full_path, 'rb') as pdf_file:
+        email.attach('sales_order_receipt.pdf', pdf_file.read(), 'application/pdf')
+    
+    # Send the email
+    email.send()
+
+    return "PDF sent via Email successfully."
+
+
+from config.settings import MEDIA_ROOT, MEDIA_URL
+import requests
```

Comment:
Move these imports top of the file, these shouldn't be in the middle

Reply:
Moved